numpy/core/include/numpy/experimental_dtype_api.h

```diff
@@ -369,6 +369,7 @@ typedef int (get_reduction_initial_function)(
 #define NPY_DT_ensure_canonical 6
 #define NPY_DT_setitem 7
 #define NPY_DT_getitem 8
+#define NPY_DT_PyArray_ArrFuncs_compare 9
```

Comment:
Ah, you are putting the slots of arrfuncs explicitly, not all of them in one.  Yeah, maybe that is better, since hopefully we don't need too many.  Compare is one, the sorting functions explicitly would make sense (but maybe it is actually OK if force whoever wants it to implement it "properly").

`arange` might be another annoying one here (which is called `fill`).